sensors/simulated/sensorhub-driver-fakegps/src/main/java/org/sensorhub/impl/sensor/fakegps/FakeGpsOutput.java

```diff
@@ -445,7 +446,7 @@ protected void start()
             // update one of the routes (randomly picked)
             if (System.currentTimeMillis() - lastApiCallTime > updatePeriod)
             {
-                int idx = (int)Math.round(Math.random() * (routes.size()-1));
+                int idx = Math.round(random.nextFloat() * (routes.size()-1));
```

Comment:
Not that it matters, since this is a test thing any way, but using `Math.round` doesn't choose the routes with uniform probability. Route `0` has a higher probability, and the last route has a lower probability of being chosen.

The "right" way is to use `random.nextInt(routes.size())`.